heed/src/env.rs

```diff
@@ -830,6 +833,59 @@ mod tests {
             .unwrap();
     }
 
+    #[test]
+    #[cfg(unix)]
+    fn open_env_with_named_path_symlink() {
+        let dir = tempfile::tempdir().unwrap();
+        let dir_symlink = tempfile::tempdir().unwrap();
+
+        let env_name = dir.path().join("babar.mdb");
+        let symlink_name = dir_symlink.path().join("babar.mdb.link");
+        fs::create_dir_all(&env_name).unwrap();
+
+        let env = EnvOpenOptions::new()
+            .map_size(10 * 1024 * 1024) // 10MB
+            .open(&env_name)
+            .unwrap();
+        assert_eq!(env_name, env.path());
+
+        std::os::unix::fs::symlink(&env_name, &symlink_name).unwrap();
+
+        let env = EnvOpenOptions::new()
+            .map_size(10 * 1024 * 1024) // 10MB
+            .open(&symlink_name)
+            .unwrap();
+
+        // The path is recycle from the first opening of the env.
+        assert_eq!(env_name, env.path());
+    }
+
+    // Unix only since managing to support *moving* directory on windows was too
+    // much pain. By moving we mean not copy-recreating, just changing name.
+    // https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-movefileexa
+    #[test]
+    #[cfg(unix)]
+    fn open_env_with_named_path_rename() {
+        let dir = tempfile::tempdir().unwrap();
+        let env_name = dir.path().join("babar.mdb");
+        fs::create_dir_all(&env_name).unwrap();
+
+        let env = EnvOpenOptions::new()
+            .map_size(10 * 1024 * 1024) // 10MB
+            .open(&env_name)
+            .unwrap();
+        assert_eq!(env_name, env.path());
+
+        let env_renamed = dir.path().join("serafina.mdb");
+        std::fs::rename(&env_name, &env_renamed).unwrap();
```

Comment:
for the windows part maybe the little UNIX api? https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/rename-wrename?view=msvc-170